src/view/map_view.py

```diff
@@ -0,0 +1,105 @@
+import json
+import os
+from typing import Any
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from matplotlib.animation import FuncAnimation
+from PIL import Image
+
+
+class ViewMap:
+    def __init__(self, map_name: str, csv_file_path: str):
+        # Get the directory of the current script
+        self.base_dir = os.path.dirname(os.path.abspath(__file__))
+
+        self.map_name = map_name
+        self.csv_file_path = os.path.join(self.base_dir, csv_file_path)
+        self.image_file_path = os.path.join(self.base_dir, "..", "assets", "maps", map_name, "display_icon.png")
+        self.json_file_path = os.path.join(self.base_dir, "..", "data", "maps", f"{map_name}.json")
+
+        self.selected_map = self.load_map_data()
+        self.map_img = self.load_map_image()
+        self.current_game = self.load_csv_data()
+
+    def load_map_image(self) -> Image.Image:
+        if not os.path.isfile(self.image_file_path):
+            raise FileNotFoundError(f"Map image not found: {self.image_file_path}")
+        try:
+            return Image.open(self.image_file_path)
+        except OSError as e:
+            raise RuntimeError(f"Failed to load map image: {e}")
+
+    def load_map_data(self) -> dict[str, Any]:
+        if not os.path.isfile(self.json_file_path):
+            raise FileNotFoundError(f"Map data not found: {self.json_file_path}")
+        try:
+            with open(self.json_file_path) as file:
+                return json.load(file)
+        except (OSError, json.JSONDecodeError) as e:
+            raise RuntimeError(f"Failed to load map data: {e}")
+
+    def load_csv_data(self) -> pd.DataFrame:
+        if not os.path.isfile(self.csv_file_path):
+            raise FileNotFoundError(f"CSV file not found: {self.csv_file_path}")
+        try:
+            return pd.read_csv(self.csv_file_path)
+        except (OSError, pd.errors.EmptyDataError, pd.errors.ParserError) as e:
+            raise RuntimeError(f"Failed to load CSV data: {e}")
+
+    def convert_coordinates(self, x: float, y: float, map_data: dict[str, Any]) -> tuple[float, float]:
```

Comment:
instead of using map_data, use GameMap class which has the attributes you need

Reply:
Solved